src/nvim/api/private/dispatch.c

```diff
@@ -32,14 +32,20 @@ static void msgpack_rpc_add_method_handler(String method,
 /// @param name API method name
 /// @param name_len name size (includes terminating NUL)
 MsgpackRpcRequestHandler msgpack_rpc_get_handler_for(const char *name,
-                                                     size_t name_len)
+                                                     size_t name_len,
+                                                     Error *error)
 {
   String m = { .data = (char *)name, .size = name_len };
   MsgpackRpcRequestHandler rv =
     map_get(String, MsgpackRpcRequestHandler)(methods, m);
 
   if (!rv.fn) {
-    rv.fn = msgpack_rpc_handle_missing_method;
+    String method_name = m.size > 0 ?
```

Comment:
can use `m.size > 0 ? m : cstr_as_string("<empty>")` and remove the freeing of this string variable.

Reply:
but wouldnt i still need to free the allocation from `cstr_as_string("<empty>")` ?

Reply:
`cstr_as_string` doesn't copy, that's the whole difference compared to `cstr_to_string`

Reply:
ah sorry; i used `cstr_to_str` and had the leak; i did not see the "as"! ok i will commit tomorrow!

Reply:
Not related. #8721 will fix this.

Reply:
Ah Okay, can i change this too [RDY]?